spec/models/project_spec.rb

```diff
@@ -333,688 +1108,799 @@
-      Project.to_finish
+      it "should not have created a notification for the project" do
+        expect(ProjectNotification).to_not have_received(:notify_once)
+      end
     end
   end
 
-  describe ".expired" do
+  describe ".order_by" do
     before do
-      @p = create(:project, online_days: 1, online_date: Time.now - 2.days)
-      create(:project, online_days: 1)
+      create(:project, name: 'a_goal_1000', goal: 1000)
+      create(:project, name: 'b_goal_400', goal: 400)
+      create(:project, name: 'c_goal_10', goal: 10)
     end
-    subject{ Project.expired}
-    it{ is_expected.to eq([@p]) }
-  end
 
-  describe ".not_expired" do
-    before do
-      @p = create(:project, online_days: 1)
-      create(:project, online_days: 1, online_date: Time.now - 2.days)
-    end
-    subject{ Project.not_expired }
-    it{ is_expected.to eq([@p]) }
-  end
+    subject { Project.order_by(order_by_query).map(&:name) }
 
-  describe ".expiring" do
-    before do
-      @p = create(:project, online_date: Time.now, online_days: 13)
-      create(:project, online_date: Time.now, online_days: 1, online_date: Time.now - 2.days)
-    end
-    subject{ Project.expiring }
-    it{ is_expected.to eq([@p]) }
-  end
+    context "when the order by query is valid" do
+      context "and order by goal" do
+        context "and it is ascending" do
+          let(:order_by_query) { "goal asc" }
 
-  describe ".not_expiring" do
-    before do
-      @p = create(:project, online_days: 15)
-      create(:project, online_days: 1, online_date: Time.now - 2.days)
-    end
-    subject{ Project.not_expiring }
-    it{ is_expected.to eq([@p]) }
-  end
+          it { is_expected.to contain_exactly('c_goal_10', 'b_goal_400', 'a_goal_1000') }
+        end
 
-  describe ".recent" do
-    before do
-      @p = create(:project, online_date: (Time.now - 4.days))
-      create(:project, online_date: (Time.now - 15.days))
-    end
-    subject{ Project.recent }
-    it{ is_expected.to eq([@p]) }
-  end
+        context "and it is descending" do
+          let(:order_by_query) { "goal desc" }
 
-  describe "send_verify_moip_account_notification" do
-    before do
-      @p = create(:project, state: 'online', online_date: DateTime.now, online_days: 3)
-      create(:project, state: 'draft')
-    end
+          it { is_expected.to contain_exactly('a_goal_1000', 'b_goal_400', 'c_goal_10') }
+        end
+      end
 
-    it "should create notification for all projects that is expiring" do
-      expect(ProjectNotification).to receive(:notify_once).
-        with(:verify_moip_account, @p.user, @p, {from_email: CatarseSettings[:email_payments]})
-      Project.send_verify_moip_account_notification
+      context "and order by name" do
+        context "and it is ascending" do
+          let(:order_by_query) { "name asc" }
+
+          it { is_expected.to contain_exactly('a_goal_1000', 'b_goal_400', 'c_goal_10') }
```

Comment:
The `contain_exactly` won't verify the order, just if the elements are contained.
To test the order you can do like this:

```ruby
it { is_expected.to eq(['a_goal_1000', 'b_goal_400', 'c_goal_10']) }
```